cmd/proxy/actions/filter.go

```diff
@@ -0,0 +1,59 @@
+package actions
+
+import (
+	"net/http"
+	"net/url"
+
+	"strings"
+
+	"github.com/bketelsen/buffet"
+	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/module"
+	"github.com/gomods/athens/pkg/paths"
+)
+
+func newFilterMiddleware(mf *module.Filter) buffalo.MiddlewareFunc {
+	const op errors.Op = "actions.FilterMiddleware"
+
+	return func(next buffalo.Handler) buffalo.Handler {
+		return func(c buffalo.Context) error {
+			sp := buffet.SpanFromContext(c).SetOperationName("filterMiddleware")
+			defer sp.Finish()
+			mod, err := paths.GetModule(c)
+
+			if err != nil {
+				// if there is no module the path we are hitting is not one related to modules, like /
+				return next(c)
+			}
+
+			// not checking the error. Not all requests include a version (i.e. list requests)
+			version, _ := paths.GetVersion(c)
```

Comment:
on err please go next(c)

Reply:
Discussed offline: will keep it here as we want List endpoint follow the same flow